doc/docs/deploy-manage/deploy/environment-variables.md

```diff
@@ -0,0 +1,166 @@
+can transmit the required configuration directly to your application.
+
+In Pachyderm, you can define the following types of environment
+variables:
+
+* `pachd` environment variables that define parameters for your
+Pachyderm daemon container.
+
+* Pachyderm worker environment variables that define parameters
+on the Kubernetes pods that run your pipeline code.
+
+You can configure environment variables as you typically do
+in a Linux environment. For example, to configure a `PACH_JOB_ID`
+in bash, run:
+
+```bash
+$ export PACH_JOB_ID=<ID>
+```
+
+You can access all the variables in the Pachyderm manifest that
+is generated when you run `pachctl deploy` with the --dry-run`
+flag.
+
+!!! note "See also"
+    [Deploy Pachyderm](../../../getting_started/local_installation/#deploy-pachyderm)
+
+## `pachd` Environment Variables
+
+You can find the list of `pachd` environment variables in the
+`pachd` manifest by running the following command:
+
+```bash
+$ kubectl get deploy pachd -o yaml
+```
+
+The following tables list all the `pachd`
+environment variables.
+
+**Global Configuration**
+
+| Environment Variable   | Default Value     | Description |
+| ---------------------- | ----------------- | ----------- |
+| `ETCD_SERVICE_HOST`    | N/A               | The host on which the etcd service runs. |
+| `ETCD_SERVICE_PORT`    | N/A               | The etcd port number.                    |
+| `PPS_WORKER_GRPC_PORT` | `80`              | The GRPs port number.                    |
+| `PORT`                 | `650`             | The `pachd` port number. |
+| `PPROF_PORT`            | `651`             | The port number for the `pprof` tool.     |
+| `HTTP_PORT`             | `652`             | The HTTP port number.   |
+| `PEER_PORT`             | `653`             | The port for pachd-to-pachd communication. |
+| `PPS_ETCD_PREFIX`        | `pachyderm_pps`   | ???  |
+| `NAMESPACE`            | `deafult`         | The namespace in which Pachyderm is deployed. |
+| `PACH_ROOT`          | `/pach`           | ??? |
+
+**pachd Configuration**
+
+| Environment Variable   | Default Value     | Description |
+| ---------------------- | ----------------- | ----------- |
+| `NUM_SHARDS`           | `32`              | The max number of `pachd` pods that can run in a single cluster. |
+| `STORAGE_BACKEND`      | `"`               | The storage backend defined for the Pachyderm cluster. By default, local storage is configured. |
+| `STORAGE_HOST_PATH`    | `"`               | The host path to storage.                    |
+| `ETCD_PREFIX`          | `"`               | A prefix for the etcd cluster.  |???
+| `PFS_ETCD_PREFIX`      | `pachyderm_pfs`   | The ???  - Should we expose this in the docs
+| `PACHYDERM_AUTH_ETCD_PREFIX` | `pachyderm_auth` |   ???? Should we expose this in the docs?
+| `PACHYDERM_ENTERPRISE_ETCD_PREFIX` | `pachyderm_enterprise` | ???? should we expose in the docs
+| `KUBERNETES_PORT_443_TCP_ADDR` | N/A   | A Kubernetes TCP address. |
```

Comment:
This is automatically exported by Kubernetes (i.e. users can only read it, not write it) and gives users the address for talking to Kubernetes's API, if they want to do that from their user code.

Just for reference, variables like `XYZ_PORT_123_TCP_ADDR` are generally kubernetes service variables: https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables